scmrepo/git/__init__.py

```diff
@@ -260,6 +260,29 @@ def get_fs(self, rev: str):
 
         return GitFileSystem(scm=self, rev=rev)
 
+    @classmethod
+    def init(
+        cls, path: str, bare: bool = False, _backend: str = None
+    ) -> "Git":
+        for name, backend in GitBackends.DEFAULT.items():
+            if _backend and name != _backend:
+                continue
+            try:
+                backend.init(path, bare=bare)
+                # TODO: reuse created object instead of initializing a new one.
+                return cls(path)
+            except NotImplementedError:
+                pass
+        raise NoGitBackendError("init")
+
+    def add_commit(
```

Comment:
We don't have `scm_gen`, so I added this `add_commit` to simplify tests. It's questionable, but `scmrepo` should also aim to provide higher-level APIs than underlying backends.